plaidml/edsl/tests/edsl_test.cc

```diff
@@ -2024,6 +2024,25 @@ TEST_F(CppEdsl, LayerUnusedOperand) {
   runProgram(program);
 }
 
+TEST_F(CppEdsl, BadDataType) {
+  // auto A = Placeholder(DType::BFLOAT16, {10, 20});
+  // Tensor O = A + A;
+  // EXPECT_ANY_THROW({ makeProgram("BadDataType", {A}, {O}); });
+  EXPECT_ANY_THROW({ auto A = Placeholder(DType::BFLOAT16, {10, 20}); });
```

Comment:
If BFLOAT16 isn't yet supported, let's just drop it from the enum.

Reply:
At that point, do you think it's worth keeping the check for the "INVALID" dtype, or is it safe to assume a user isn't going to try and use that and expect it to work?

Reply:
Perhaps we could check for the invalid dtype deeper, in the ffi. Generally, I try to do validation inside the ffi so that every language benefits from it.

Reply:
Actually, not in the ffi, more like in the construction of an AST node.